src/GSoft.Extensions.Mongo.ApplicationInsights/ApplicationInsightsEventSubscriberFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.Options;
+using MongoDB.Driver.Core.Events;
+
+namespace GSoft.Extensions.Mongo.ApplicationInsights;
+
+internal sealed class ApplicationInsightsEventSubscriberFactory : IMongoEventSubscriberFactory
+{
+    private readonly IOptionsMonitor<MongoClientOptions> _optionsMonitor;
+    private readonly TelemetryClient? _telemetryClient;
+
+    public ApplicationInsightsEventSubscriberFactory(IOptionsMonitor<MongoClientOptions> optionsMonitor, TelemetryClient? telemetryClient = null)
+    {
+        this._telemetryClient = telemetryClient;
+        this._optionsMonitor = optionsMonitor;
+    }
+
+    public IEnumerable<IEventSubscriber> CreateEventSubscribers(string clientName)
```

Comment:
In this context, it is okay to return an empty `IEnumerable` if the client would be `null`?

Reply:
yield require less code and eliminates the need of using Enumerable.Empty<> and creating a new List/array when the telemetryclient is not null.